### PR TITLE
feat(service-portal): finance schedule new module 

### PR DESCRIPTION
**Pull request by disaerna**
# Payment Schedule added for Service Portal

## What

* Add new api calls for finance schedule
- _get list of all payment schedules_
- _get detailed view for payment schedule_
* Add new module called "finance-schedule"
- _new model added so it can be feature flagged_
* Add translation namespace
* Refactor finance expandable row and move plus sign in front
* Refactor some utils code to service-portal core library 

## Why

New feature -  Greiðsluáætlanir 📄  💰 

## Screenshots / Gifs

Screenshot 2022-04-01 at 08 49 38

Screenshot 2022-04-01 at 08 48 30

Screenshot 2022-04-01 at 08 49 27

Screenshot 2022-04-01 at 08 49 15

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/service-portal/finance/src/components/FinanceDebtStatus/FinanceDebtStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  GridContainer,
+} from '@island.is/island-ui/core'
+
+interface Props {
+  debtStatusData: any
+}
+
+const FinanceDebtStatus: FC<Props> = ({ debtStatusData }) => {
+  const { formatMessage } = useLocale()
+  console.log('DebtStatusData: ', debtStatusData)
```
Comment (thordurhhh) [inline libs/service-portal/finance/src/components/FinanceDebtStatus/FinanceDebtStatus.tsx:19]: 😱

File: libs/service-portal/finance/src/index.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { gql } from '@apollo/client'
```
Comment (thordurhhh) [inline libs/service-portal/finance/src/index.ts:1]: Not used?

File: libs/service-portal/finance-schedule/src/components/FinanceScheduleTable/FinanceScheduleTableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  const [
+    getPaymentScheduleById,
+    { loading, error, ...detailsQuery },
+  ] = useLazyQuery(GET_FINANCE_PAYMENT_SCHEDULE_BY_ID)
+
+  const paymentDetailData: Array<DetailedSchedule> =
+    detailsQuery?.data?.getPaymentScheduleById.myDetailedSchedules
+      .myDetailedSchedule || []
+
+  const getType = (type: string) => {
```
Comment (thordurhhh) [inline libs/service-portal/finance-schedule/src/components/FinanceScheduleTable/FinanceScheduleTableRow.tsx:50]: Can we translate this, and look at an alternative default value 🚀

File: libs/service-portal/finance/src/components/FinanceDebtStatus/FinanceDebtStatus.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        <GridContainer position="relative">
+          <GridRow>
+            <GridColumn span="5/12" offset="1/12">
+              <Box
+                display="flex"
+                flexDirection="row"
+                justifyContent="spaceBetween"
+              >
+                <Text fontWeight="semiBold">
+                  {'Staða við ríkissjóð og stofnanir'}
```
Comment (thordurhhh) [inline libs/service-portal/finance/src/components/FinanceDebtStatus/FinanceDebtStatus.tsx:32]: Few translations missing in this file 🙏 

l: 43, 55, 65

File: libs/service-portal/finance/src/components/FinanceDebtStatus/FinanceDebtStatus.tsx (hunk 1 above)
Comment (disaerna) [reply libs/service-portal/finance/src/components/FinanceDebtStatus/FinanceDebtStatus.tsx]: 🤭 Removed

File: libs/service-portal/finance/src/index.ts (hunk 2 above)
Comment (disaerna) [reply libs/service-portal/finance/src/index.ts]: No - removed ✅

**Review by thordurhhh — APPROVED** [review]
Nice 🚀

**Review by saevarma — APPROVED** [review]
Those 4 core files LGTM

**Pull request merged**